### ISSUE TITLE
Disable automatic updates for OS major/minor releases

### ISSUE THREAD

**Issue by MrBatschner** [gardener/gardener#6365]
**How to categorize this issue?**

/area robustness
/kind enhancement

**What would you like to be added**:
Gardeners maintenance controller would currently upgrade the OS of worker nodes of those shoots that have automatic patch management enabled to the latest release as soon as it is classified as `supported` in the CloudProfile. 
For new Kubernetes versions, Gardener would honour the major.minor.patchlevel semantic and would do automatic upgrades only for patch-releases but not for changes in the major or minor version. OS versions on the other hand would get upgraded as soon as a higher version gets introduced - independent from the major.minor.patchlevel semantic.
**TL;DR:** The update candidate logic for OS releases whose version number follows semver should be like the logic for Kubernetes versions in that only patch releases get automatically upgraded.

**Why is this needed**:
New operating systems might introduce changes that are incompatible to previous releases and that could potentially break existing workloads. For patch-releases of an operating system, automatic upgrades are fine as we should expect OS vendors not to introduce incompatible changes into them. For new major/minor releases (such as Garden Linux 318 to 576 or SuSE cHost 15.3 to 15.4) changes can be more drastic such as a completely different Linux kernel, changes in container runtimes, etc. 
Upgrading the worker nodes of a Shoot to a new OS release with such potentially extensive changes should be a deliberate decision of the user and not something that will automatically triggered.

**Comment by rfranzke** [comment]
@danielfoehrKn @vlerenc Do you have an opinion about this one? IIRC you were mostly involved in the discussions back then when the whole auto-update/classification features were developed.

**Comment by vlerenc** [comment]
@rfranzke Sure, I love what @MrBatschner is writing:

> TL;DR: The update candidate logic for OS releases whose version number follows semver should be like the logic for Kubernetes versions in that only patch releases get automatically upgraded.

Absolutely. It was always suboptimal that we didn't have OS semver versioning and if we now have, then we should handle it similarly.

There are a few (?) subtleties, though:
- Are the test suites set up in this way, to validate all patch updates for their respective major/minor versions resp. dealing with multiple major/minor versions in the first place?
- Different than for Kubernetes versions, the Gardener Dashboard doesn't make this a simple operation (API/`shoot` spec is fine, though).
- Documentation must be overhauled accordingly (https://wiki.one.int.sap/wiki/display/Kubernetes/Versioning+Policy - something else?)
- ...

Maybe we start at an updated/new version of https://github.com/gardener/gardener/blob/77e09dd522d7bfb682cc35d0be437201b606061b/docs/proposals/05-versioning-policy.md, which is how what we have today all started? Just to understand the exact intent in more detail?

**Comment by rfranzke** [comment]
Should we close this in favor of #8013?
